.github/workflows/ci.yml

```diff
@@ -252,13 +267,77 @@ jobs:
           name: nuget-packages
           path: ./Bin/Release/ILGPU*.${{ steps.package.outputs.version }}.*nupkg
 
-    outputs:
-      version: ${{ steps.version.outputs.version }}
+  # Create the Samples zip file, replacing ILGPU project references with equivalent NuGet package references.
+  package-samples:
+    needs: check-version
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup the latest .NET 5 SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.x
+
+      # Change the ILGPU project references to NuGet package references
+      - name: Update Sample References
+        id: package
+        run: |
+          $version = "${{ needs.check-version.outputs.version }}"
+
+          if (-not ("${{ github.ref }}" -like "refs/tags/v*")) {
+            $suffix = "preview-$(Get-Date -Format yyyyMMddHHmmss -AsUTC)-$(git rev-parse --short HEAD)"
+            $params = "--version-suffix", $suffix
+            $version = "$version-$suffix"
+          }
```

Comment:
possible improvement: I believe this could actually be refactored to be part of the `check-version` job so that we don't duplicate the logic between `package-library` and `package-samples`.
It would compute the version, the suffix, and the full version. Then `package-library` can build its `$params` based on it, `package-samples` would only need the full version, and `publish-release` would only need the version.

Reply:
Done.